### DISCUSSION TITLE
Allow to pick which components should be deployed

### DISCUSSION THREAD

**Discussion by renepajta** [Azure/data-domain-batch#76] [Ideas] [answered]
At the moment, data product (batch) deploys various components (Synapse, Cosmos, SQL flavor) which may not be necessarily needed by the Data Product team. 

Could we make the deployment of individual services optional?

**Comment by marvinbuss** [comment, marked as answer]
Of course we can. Which services would you like to make optional?

Please keep in mind that the Cosmos account is free, if no databases are created. Same is true for the Synapse workspace.

**Reply by renepajta** [reply]
In my opinion, the end goal should be to deploy only the necessary services and if the team needs other services later, just redeploy the template with new resources. 

Based on this, I would keep everything optional and let the team decide what they need.

**Reply by marvinbuss** [reply]
We are trying to be as prescriptive as possible. Therefore, making all resources optional is not something that I would like to avoid. I strongly believe that some of the pieces are essential for any successful batch data processing project. But you are right, some of the services can be seen as optional (Database, Synapse vs. Data Factory, etc.).

Also, redeployment may not work, as some of the resource names are already taken, if you redeploy with the same prefix.

**Reply by renepajta** [reply]
I think we will be wiser after few customer deployments what is preference and which services can be marked as optional.

**Reply by marvinbuss** [reply]
Will do so and keep track of this. I will add this to the backlog and work on the necessary changes later this month. Progress can be tracked here: https://github.com/Azure/data-product-batch/issues/80
